source/server/api_listener_impl.cc

```diff
@@ -44,5 +51,12 @@ Http::ApiListenerOptRef HttpApiListener::http() {
   return Http::ApiListenerOptRef(std::ref(*http_connection_manager_));
 }
 
+void HttpApiListener::shutdown() {
+  // The Http::ConnectionManagerImpl is a callback target for the read_callback_.connection_. By
+  // raising connection closure, Http::ConnectionManagerImpl::onEvent is fired. In that case the
+  // Http::ConnectionManagerImpl will reset any ActiveStreams it has.
+  read_callbacks_.connection_.raiseConnectionEvent(Network::ConnectionEvent::LocalClose);
```

Comment:
Pedantically, I would probably use RemoteClose here, since it's the "client" that is closing.

Reply:
I was thinking local because the close happens because envoy is terminating. But I guess envoy is terminating because the "client" decided to shut it down. 

I can change to remote.